src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -189,11 +198,17 @@ public bool Equals(EmitOptions other)
                 this.RuntimeMetadataVersion == other.RuntimeMetadataVersion &&
                 this.TolerateErrors == other.TolerateErrors &&
                 this.IncludePrivateMembers == other.IncludePrivateMembers &&
-                this.Instrument == other.Instrument;
+                this.InstrumentationKinds == other.InstrumentationKinds;
         }
 
         public override int GetHashCode()
         {
+            var instrumentationKindHash = 0;
+            foreach (var instrumentationKind in InstrumentationKinds)
+            {
+                instrumentationKindHash = Hash.Combine((int)instrumentationKind, instrumentationKindHash);
+            }
```

Comment:
Note, we have Hash.CombineValues and SequenceEquals helpers, see how they are used in ParseOptions.